test/e2e/create_update_delete_test.go

```diff
@@ -143,8 +143,7 @@ data:
 	appName := "test-create-update-delete-prev-app"
 	prevAppName := "test-create-update-delete-prev-app-old"
 	cleanUp := func() {
-		kapp.Run([]string{"delete", "-a", appName})
-		kapp.Run([]string{"delete", "-a", prevAppName})
+		kapp.Run([]string{"delete", "-a", appName, "--prev-app", prevAppName})
```

Comment:
I may have missed some tests, but I think this will always delete `appName` as in all the tests we are using the `--prev-app` flag with deploy which will migrate the app to `appName`. 
(or I may have missed some test)

Other than this, everything else looks good to me.

Reply:
Cool, I added some explicit tests and removed them from the cleanup.